### PR TITLE
Doc warning for pure java jai / native jai incompatibility [GEOS-7554]

### PR DESCRIPTION
**Pull request by jodygarnett**
Please see: https://osgeo-org.atlassian.net/browse/GEOS-7554

### CODE REVIEW

File: doc/en/user/source/production/java.rst [hunk 1]
```diff
@@ -17,19 +17,31 @@ Tested:
...
+The `Java Image I/O Technology <http://docs.oracle.com/javase/6/docs/technotes/guides/imageio/index.html>`__ (ImageIO) is used for  raster reading and writing. ImageIO affects both WMS and WCS for reading raster data, and is very useful (even if there is no raster data involved) for WMS output as encoding is required when writing PNG/GIF/JPEG images.
 
-Native extensions are available for Windows, Linux and Solaris (32 and 64 bit systems).  They are, however, not available for OS X.
+By default, GeoServer ships with the "pure java" version of JAI, but **for better performance JAI and ImageIO are available as "java extensions" to be installed into your JDK/JRE**. Native JAI and ImageIO extensions are available for Windows, Linux and Solaris (32 and 64 bit systems).  They are, however, not available for OS X.
```
Comment (simboss) [inline doc/en/user/source/production/java.rst]: They are not available in Win 64

File: doc/en/user/source/production/java.rst [hunk 2]
```diff
@@ -17,19 +17,31 @@ Tested:
...
 
+.. warning:: These native installers include a copy of JAI and JAI ImageIO as a "java extension" in your JRE. This may conflict with the pure java copy of JAI and JAI ImageIO included in your GeoServer ``libs`` folder - producing "class cast exceptions" preventing your application server from starting GeoServer.
```
Comment (simboss) [inline doc/en/user/source/production/java.rst]: I would clarify that they also install JARs in the JRE in addition to native libs.

File: doc/en/user/source/production/java.rst [hunk 3]
```diff
@@ -17,19 +17,31 @@ Tested:
...
+   
+   After installing the native JAI and ImageIO extensions remove the pure java implementation from your GeoServer instances ``WEB-INF/lib`` folder::
```
Comment (simboss) [inline doc/en/user/source/production/java.rst]: ... if you keep the JAI and ImageIO JARs in the JRE then remove....

**Comment by dvntucker** [comment]
Looks OK to me. Did not see any spelling mistakes (although IANACE - I Am Not A Copy-Editor), content  updates look good.

File: doc/en/user/source/production/java.rst (hunk 1 above)
Comment (tbarsballe) [inline doc/en/user/source/production/java.rst]: While technically correct, "They are, however, " is rather awkward. Drop the ", however," altogether, or use "However, they are".

File: doc/en/user/source/production/java.rst [hunk 4]
```diff
@@ -87,7 +99,7 @@ Please refer to the `GeoTools page on JAI installation <http://docs.geotools.org
 GeoServer cleanup
 `````````````````
 
-Once the installation is complete, you may optionally remove the original JAI files from the GeoServer instance::
+Once the installation is complete, you may optionally remove the original JAI files from the GeoServer instances ``WEB-INF/lib`` folder::
 
    jai_core-x.y.z.jar
```
Comment (simboss) [inline doc/en/user/source/production/java.rst]: MIssing jai_codec-1.1.3.jar

File: doc/en/user/source/production/java.rst [hunk 5]
```diff
@@ -87,7 +99,7 @@ Please refer to the `GeoTools page on JAI installation <http://docs.geotools.org
 GeoServer cleanup
 `````````````````
 
-Once the installation is complete, you may optionally remove the original JAI files from the GeoServer instance::
+Once the installation is complete, you may optionally remove the original JAI files from the GeoServer instances ``WEB-INF/lib`` folder::
```
Comment (tbarsballe) [inline doc/en/user/source/production/java.rst]: Use "instance's", else it sounds like you are refering to multiple geoservers.

**Comment by simboss** [comment]
see my comments...

File: doc/en/user/source/production/java.rst (hunk 3 above)
Comment (jodygarnett) [inline doc/en/user/source/production/java.rst]: I would always recommend keeping the jars in the JRE (with the native code) so the JRE would be consistent.

I always thought removing the java extension jars would prevent the use of the native code...

File: doc/en/user/source/production/java.rst (hunk 4 above)
Comment (jodygarnett) [inline doc/en/user/source/production/java.rst]: jai_codec was further down in the original file (ie not part of the diff).

**Pull request merged**